### PR TITLE
Allow DateInput value to initialize as []

### LINKED ISSUE #5164: DateInput with range of dates doesn't work if value starts as undefined
### Expected Behavior

DateInput with desired format of mm/dd/yyyy - mm/dd/yyyy should be able to be initialized with no value. and then when a user interacts, they can select a range.

### Actual Behavior

When no value is set, the user is only ever able to select the first date of the range. After which, the Calendar closes and the end date cannot be selected.

This is because DateInput checks the type of the incoming `value` prop. If it's not an array, then it assumes this DateInput is only aiming to support single date selection. User should be able to flag `range` as a prop on DateInput to opt into the range of dates behavior OR Calendar should support an empty array `[]` for dates.

### URL, screen shot, or Codepen exhibiting the issue

https://codesandbox.io/s/dateinput-range-with-undefined-value-at-start-ei9sj?file=/index.js

### Steps to Reproduce

1. Click into DateInput
2. Select date
3. Calendar closes, cannot select a range.

### Your Environment

- Grommet version: Stable
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MacOS

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
DateInput uses a singular `value` prop to indicate how the date or dates should be passed to the underlying Calendar. This enhances DateInput to allow caller to initialize value as an empty array `[]` which is relevant if they want the DateInput to be a range of dates but they don't want any dates to be selected to start.

Some small changes were also made to the utils function that decides when the placeholder value is shown or not to ensure the placeholder is still shown when the array is empty.

#### Where should the reviewer start?
src/js/components/DateInput/DateInput.js

#### What testing has been done on this PR?
Tested in storybook on DateInput range format story. Updated value to `[]` in useState initialization.

Added jest test.

#### How should this be manually tested?
In storybook.

#### Any background context you want to provide?
DateInput with desire for range had to be initialized with a value, if not only the first date of the range was able to be selected. Caller could not start with an empty range.

#### What are the relevant issues?
Closes #5164 

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
Fixed bug in DateInput to allow value to be initialized as `[]`, which allows range of dates to start empty.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by ShimiSun — COMMENTED** [review]
Jest isn't showing love to this test

image

**Comment by taysea** [comment]
> Jest isn't showing love to this test
> 
> image

Looking into it, thanks!

File: src/js/components/DateInput/DateInput.js [hunk 1]
```diff
@@ -92,7 +92,9 @@ const DateInput = forwardRef(
         id={inline && !format ? id : undefined}
         range={range}
         date={range ? undefined : value}
-        dates={range ? [value] : undefined}
+        // when caller initializes with empty array, dates
+        // should be undefined
```
Comment (halocline) [inline src/js/components/DateInput/DateInput.js:96]: I found myself asking "why should this be undefined?" Suggesting we state the desired behavior "allow user to select both start and end"

```suggestion
        // when caller initializes with empty array, dates should be undefined 
        // allowing the user to select both begin and end of the range
```

**Review by halocline — APPROVED** [review]
Minor comment. All else looks good.

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**